misk-aws/src/main/kotlin/misk/jobqueue/sqs/QueueResolver.kt

```diff
@@ -67,7 +72,29 @@ internal class QueueResolver @Inject internal constructor(
       throw e
     }
 
-    return ResolvedQueue(q, sqsQueueName, queueUrl, region, accountId, sqs)
+    val queueMaxRetries = extractMaxReceiveCount(sqs, queueUrl)
+    return ResolvedQueue(q, sqsQueueName, queueUrl, region, accountId, sqs, queueMaxRetries)
+  }
+
+  data class RedrivePolicy(val maxReceiveCount: Int, val deadLetterTargetArn: String)
```

Comment:
I think `RedrivePolicy` is a misnomer, this doesn't control whether we have a DLQ or after how many delivery attempts a job will be put in the DLQ, it just controls the max backoff period for retrying a job

Reply:
hmmm that's actually the name of it in AWS: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html.
but if you have a suggestion on the better name please let me know!

Reply:
we are effectively fetching the policy in order to determine the maxRetryCount on the particular queue 🤔

Reply:
Ah my mistake, this attribute is actually populated on the queue when we create it, I thought this was something custom for this PR.